.github/workflows/test.yml

```diff
@@ -7,10 +7,24 @@ on:
       - pr/*
   workflow_dispatch:
 
+env:
+  CLANG_LLVM_BASE_URL: "https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.0/clang+llvm-13.0.0-x86_64"
+  ARCHIVE_EXT: "tar.xz"
+
 jobs:
 
   linux-x64:
-    runs-on: "ubuntu-20.04"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-20.04]
+        include:
+          - os: ubuntu-20.04
+            TARGET: linux-gnu-ubuntu-20.04
+            JAVA19_HOME: /tmp/deps/jdk-19
+          - os: macos-latest
+            TARGET: apple-darwin
+            JAVA19_HOME: /tmp/deps/jdk-19/jdk-19.jdk/Contents/Home
```

Comment:
Both of these test runs now show up as part of the same job called `linux-x64`. I suggest splitting this into 3 jobs instead:
1. a job that does all the setup.
2. a job for linux testing
3. a job for mac testing

Where both 2 and 3 depend on 1.  (see for instance how the JDK workflow achieves this using job outputs: https://github.com/openjdk/jdk/blob/master/.github/workflows/submit.yml#L149)

'Testing' would include the 'Build Jextract' and 'Run Tests' steps.

Reply:
Actually, it looks like that might not be that easy, since the prerequisites also have to pick a platform to run on.

In that case, I suggest just changing the name of the job to be something like `Build & Test`.

Reply:
At first, i thought about replacing one config with per-platform. The only foreseeable scenario with a per-platform config - is when we'd need to add Windows. Because eventually, we'd have to release a binary for Win as well. But as of now, one config for both Debian and macOS is a good-to-go solution.

Reply:
okay, I renamed a job to `build-and-test`